src/web/components/ClickToView.tsx

```diff
@@ -89,74 +113,115 @@ const Headline = ({
 
 const Body = ({
 	children,
-	width,
+	role,
 }: {
 	children: React.ReactNode;
-	width: number;
-}) => (
-	<div
-		className={css`
-			${width > 300 ? textSans.small() : textSans.xsmall()}
-			a {
-				${width > 300 ? textSans.small() : textSans.xsmall()}
-			}
-		`}
-	>
-		{children}
-	</div>
-);
+	role: RoleType;
+}) => {
+	const textSize = roleTextSize(role);
+	return (
+		<div
+			className={css`
+				${textSize}
+				a {
+					${textSize}
+				}
+			`}
+		>
+			{children}
+		</div>
+	);
+};
+
+const AcceptButton = ({
+	role,
+	handleClick,
+}: {
+	role: RoleType;
+	handleClick: Function;
+}) => {
+	return (
+		<div>
+			<Button
+				priority="primary"
+				size={roleButtonSize(role)}
+				icon={<SvgCheckmark />}
+				iconSide="left"
+				onClick={() => handleClick()}
+			>
+				{roleButtonText(role)}
+			</Button>
+		</div>
+	);
+};
 
-export const ClickToView = ({ children, width, height, onAccept }: Props) => {
+export const ClickToView = ({
+	children,
+	role,
+	onAccept,
+	isTracking,
+	source,
+	sourceDomain,
+}: Props) => {
 	const [showOverlay, setShowOverlay] = useState<boolean>(true);
 
 	const handleClick = () => {
 		setShowOverlay(false);
-		onAccept();
+		if (onAccept) {
+			setTimeout(() => onAccept());
+		}
 	};
 
-	if (showOverlay) {
+	let headlineText;
+	let body;
+
+	const roleWithDefault: RoleType = role || 'inline';
+
+	if (isTracking && showOverlay) {
+		if (source) {
+			headlineText = `Allow ${source} content?`;
+			body = (
+				<div>
```

Comment:
Ideally we wouldn't need this div. I think by removing the abstraction it becomes moot but in any case it could be replaced with a `<></>` fragment

Reply:
👍